src/main/webapp/js/finscholar/homepagecontroller/homepagecontroller.js

```diff
@@ -16,128 +16,25 @@
-    this.bindedNavbarOnclickHandler_ = this.handleNavbarOnclickEvent_.bind(this);
-
-    /** @private {number} */
-    this.navbarPageIndex_ = 0;
-
-    /** @private @const 
-     * {!Array<CollegeListView, ScholarshipListView, 
-     *     CollegePageView, ScholarshipPageView, ErrorPageView>} 
-     */
-    this.TEMPLATE_HANDLERS_ = [
-                               CollegeListView, 
-                               ScholarshipListView, 
-                               CollegePageView, 
-                               ScholarshipPageView, 
-                               ErrorPageView
-                               ];
-    
-    this.initJsaction_();
-
-    this.container_.innerHTML = this.getContent_();
-
-    /** @private @const {!Element} subView_*/
-    this.subView_ = /** @type {!Element} */ (googDom.getElement('content'));
-
-    this.renderPage_();
-  }
-
-  /**
-   * Sets up the event handlers for elements in the main frame.
-   * @private
-   */
-  initJsaction_() {
-    // Events will be handled for all elements under this container.
-    this.eventContract_.addContainer(
-        /** @type {!Element} */ (googDom.getElement('main')));
-    // Register the event types we care about.
-    this.eventContract_.addEvent('click');
-    this.eventContract_.addEvent('dblclick');
-    this.eventContract_.dispatchTo(
-        this.dispatcher_.dispatch.bind(this.dispatcher_));
-    this.dispatcher_.registerHandlers(
-      'homepagecontroller',  // the namespace
-      null,                  // handler object
-      {
-        // action map
-        'clickAction': this.bindedNavbarOnclickHandler_,
-        'doubleClickAction' : this.bindedNavbarOnclickHandler_,
-      });
-  }
-
-  /**
-   * @return {!googSoy.data.SanitizedHtml} The rendered HTML of the common framework.
-   * @private
-   */
-  getContent_() {
-    return homepage();
-  }
-
-  /**
-   * Handles click and double click events on navbar.
-   * @param {!jsactionActionFlow} flow Contains the data related to the action.
-   *     and more. See actionflow.js.
-   * @private
-   */
-  handleNavbarOnclickEvent_(flow) {
-    const index = flow.node().getAttribute('index');
-    this.navbarPageIndex_ = parseInt(index, 10);
-    this.renderPage_();
   }
 
   /**
-   * Render the div with id 'content' based on the click event navber buttons.
-   * @private
+   * Loads the home page view.
+   * @override
    */
-  renderPage_() {
-    // This if statement is temporarily added to enable showing the error page.
-    if (this.navbarPageIndex_ == 4) {
-      this.TEMPLATE_HANDLERS_[this.navbarPageIndex_].renderView(this.subView_, 
-        TEST_ERR_INPUT1,
-        TEST_ERR_INPUT2,
-        TEST_ERR_INPUT3);
-    } else {
-      // In actual product we'll only have this line in this function.
-      (new this.TEMPLATE_HANDLERS_[this.navbarPageIndex_]).renderView(this.subView_);
-    }
+  async renderView() {
+    super.setCurrentContent(homepage());
+    super.resetAndUpdate();
```

Comment:
Same comment as the other PR. These lines for the renderView method are identical across all subclasses. You can create a `getTemplate` method or something and override that instead.

Reply:
So I dropped the `renderView` method, and moved it to `super`. The sub classes only set the fields on `super` so that `renderView` works correctly.

Any thoughts?